### PR TITLE
WebGPU Renderer: cache GPUBindGroupLayouts

### PR DESCRIPTION
**Pull request by aardgoose**
Currently bindGroups and bindGroupLayouts are creates 1:1.  Cache layouts per bindGroup definition, and reused.

webgpu_materials_toons:

layouts without caching 404,
layouts with caching: 6

webgpu_materials_video:

Without caching creates many layouts per frame., with caching 6

### CODE REVIEW

File: src/renderers/common/BindGroup.js [hunk 1]
```diff
@@ -2,11 +2,12 @@ let _id = 0;
 
 class BindGroup {
 
-	constructor( name = '', bindings = [], index = 0 ) {
+	constructor( name = '', bindings = [], index = 0, cacheKey ) {
 
 		this.name = name;
 		this.bindings = bindings;
 		this.index = index;
+		this.cacheKey = cacheKey;
```
Comment (sunag) [inline src/renderers/common/BindGroup.js:10]: `cacheKey` seems to suggest a string, I think we could have `bindingsReference` or something similar.

File: src/renderers/common/BindGroup.js [hunk 2]
```diff
@@ -2,11 +2,12 @@ let _id = 0;
 
 class BindGroup {
 
-	constructor( name = '', bindings = [], index = 0 ) {
+	constructor( name = '', bindings = [], index = 0, cacheKey ) {
```
Comment (sunag) [inline src/renderers/common/BindGroup.js:5]: I think interessing complete `cacheKey|bindingsReference = []` to avoid undefined values.

Comment (aardgoose) [reply src/renderers/common/BindGroup.js]: Done

File: src/renderers/common/BindGroup.js (hunk 1 above)
Comment (aardgoose) [reply src/renderers/common/BindGroup.js]: done

**Comment by sunag** [comment]
Do you think we still need this workaround? https://github.com/mrdoob/three.js/pull/27447

**Comment by aardgoose** [comment]
> Do you think we still need this workaround? #27447

Sadly, we still do.

**Pull request merged**